### PR TITLE
(WIP) Notify TextInputConnection of a closed connection

### LINKED ISSUE #117771: TextInputConnection should let objects listen for IME connection closure
An IME connection is established with thee pieces: `TextInput`, a `TextInputConnection`, and a `TextInputClient`.

```dart
final myConnection = TextInput.attach(myInputClient, myConfig);
```

`TextInputConnection` has a `bool` that reports `isAttached`, but there's no way to listen for a change to this attachment.

The `TextInputClient` has a method called `connectionClosed`, which, I assume, represents a notification that the IME connection has closed. However, policies about when to open and close a connection are usually independent from the policies that apply text input messages from the platform. Therefore, the use of `TextInputConnection` should be able to exist independently from the code that implements `TextInputClient`, and this requires that `TextInputConnection` receive a notification when its connection is closed.

```
Flutter 3.7.0-1.2.pre • channel beta • https://github.com/flutter/flutter.git
Framework • revision c29b09b878 (9 days ago) • 2022-12-19 18:30:12 -0600
Engine • revision 8c99b0feaf
Tools • Dart 2.19.0 (build 2.19.0-444.2.beta) • DevTools 2.20.0
```

### PR DESCRIPTION
**Pull request by justinmc**
A TextInputConnection should ostensibly be able to know when it's being closed without relying on the implementation of a TextInputClient.

Fixes https://github.com/flutter/flutter/issues/117771

### CODE REVIEW

File: packages/flutter/lib/src/services/text_input.dart [hunk 1]
```diff
@@ -1463,10 +1463,14 @@ class TextInputConnection {
 
   /// Platform sent a notification informing the connection is closed.
   ///
-  /// [TextInputConnection] should clean current client connection.
+  /// [TextInputConnection] should clean the current client connection.
```
Comment (matthew-carroll) [inline packages/flutter/lib/src/services/text_input.dart:1466]: The connection should also report this information, and allow a listener to listen for it. It's not the connection, itself, that really needs this info, it's the object that app/package object that owns the connection that needs this info.

**Comment by matthew-carroll** [comment]
@justinmc can you let me know why this was closed?

**Comment by justinmc** [comment]
Just because I was cleaning up PRs and couldn't remember exactly how to finish this one...  I'll come back to it since you still need it though.  Let's talk tomorrow at the Flutter Forward meetup if we have time!  Otherwise I'll remind myself to take another look later this week.

**Comment by goderbauer** [comment]
(Triage) @justinmc Did you still have plans for this PR?

**Comment by justinmc** [comment]
@matthew-carroll Is this PR still useful to you?  What information were you looking to get when you wrote [this comment](https://github.com/flutter/flutter/pull/118283#discussion_r1067567819)?

Now in this PR the relationship is `TextInput` has a `TextInputConnection` which has a `TextInputClient`.

**Pull request closed without merging**